CodeModel/src/main/java/org/openzen/zenscript/codemodel/statement/BlockStatement.java

```diff
@@ -74,6 +75,10 @@ public Statement transform(ExpressionTransformer transformer, ConcatMap<LoopStat
 
 	@Override
 	public Optional<TypeID> getReturnType() {
+		if(statements.length == 0) {
```

Comment:
I wasn't sure if this check should be a special-case check here or further below if the `collect` List is empty?